.githooks/pre-commit

```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
+
+set -eu -o pipefail
+
+echo "Pre-commit hook running"
+
+echo "Running gofmt..."
+gofmt -w -s .
+
+echo "Running go mod tidy..."
+go mod tidy
+
+if ! command -v golangci-lint &> /dev/null ; then
```

Comment:
`&>` redirection is not supported in `/bin/sh`.
Either change the shebang to `#!/bin/bash`
Or change this line, then it will be compatible for both `sh` and `bash`:
```suggestion
if [ ! command -v golangci-lint >/dev/null 2>&1 ]; then
```

Also it's a good practice to just wrap the if stuff in braces.